runware/base.py

```diff
@@ -1367,6 +1369,193 @@ def check(resolve: Callable, reject: Callable, *args: Any) -> bool:
+        self._addVideoImages(request_object, requestVideo)
+        self._addProviderSettings(request_object, requestVideo)
+        return request_object
+
+    def _addOptionalVideoFields(self, request_object: Dict[str, Any], requestVideo: IVideoInference) -> None:
+        optional_fields = [
+            "outputType", "outputFormat", "outputQuality", "uploadEndpoint",
+            "includeCost", "negativePrompt", "fps", "steps", "seed",
+            "CFGScale", "seedImage", "duration", "width", "height",
+        ]
+
+        for field in optional_fields:
+            value = getattr(requestVideo, field, None)
+            if value is not None:
+                request_object[field] = value
+
+    def _addVideoImages(self, request_object: Dict[str, Any], requestVideo: IVideoInference) -> None:
+        if requestVideo.frameImages:
+            frame_images_data = []
+            for frame_item in requestVideo.frameImages:
+                frame_images_data.append({k: v for k, v in asdict(frame_item).items() if v is not None})
+            request_object["frameImages"] = frame_images_data
+
+        if requestVideo.referenceImages:
+            request_object["referenceImages"] = requestVideo.referenceImages
+
+    def _addProviderSettings(self, request_object: Dict[str, Any], requestVideo: IVideoInference) -> None:
+        if not requestVideo.providerSettings:
+            return
+        provider_dict = requestVideo.providerSettings.to_request_dict()
+        if provider_dict:
+            request_object["providerSettings"] = provider_dict
+
+    async def _handleInitialVideoResponse(self, task_uuid: str, number_results: int) -> List[IVideo]:
+        lis = self.globalListener(taskUUID=task_uuid)
+
+        def check_initial_response(resolve: callable, reject: callable, *args: Any) -> bool:
+            response_list = self._globalMessages.get(task_uuid, [])
+
+            if not response_list:
+                return False
+
+            response = response_list[0]
+
+            if response.get("code"):
+                raise RunwareAPIError(response)
+
+            if response.get("status") == "success":
+                del self._globalMessages[task_uuid]
+                resolve([response])
+                return True
+
+            del self._globalMessages[task_uuid]
+            resolve("POLL_NEEDED")
+            return True
+
+        try:
+            initial_response = await getIntervalWithPromise(
+                check_initial_response,
+                debugKey="video-inference-initial",
+                timeOutDuration=30000
+            )
+        finally:
+            lis["destroy"]()
+
+        if initial_response == "POLL_NEEDED":
+            return await self._pollVideoResults(task_uuid, number_results)
+        else:
+            return instantiateDataclassList(IVideo, initial_response)
+
+    async def _pollVideoResults(self, task_uuid: str, number_results: int) -> List[IVideo]:
+        max_polls = 240
```

Comment:
it's 720 seconds to give up. I think WAN generates stuff for 600 seconds. I wonder is it enough? cc @bghira

Reply:
i don't think hardcoding it is a good idea at all